addons/stock/models/stock_picking.py

```diff
@@ -248,7 +248,9 @@ def _onchange_sequence_code(self):
     @api.constrains('default_location_dest_id')
     def _check_default_location(self):
         for record in self:
-            if record.code == 'mrp_operation' and record.default_location_dest_id.scrap_location:
+            if record.active and record.code == 'mrp_operation' and record.default_location_dest_id.scrap_location:
+                if record.active:
```

Comment:
We should just also block the behavior for archived `picking_type` records too. Also, I just tested in the v16 runbot and it's already blocked so I don't understand the use case
![image](https://github.com/user-attachments/assets/6791e336-a956-4567-b4d2-ec61043fa08b)